insights/parsers/softnet_stat.py

```diff
@@ -80,13 +81,15 @@ def per_cpu_nstat(self, key):
         Returns:
             (list): Column states per cpu.
         """
-        nstats = []
-        for each_cpu in self.nstat_per_cpu:
-            value = each_cpu.get(key, '_')
-            nstats.append(int(value) if value.isdigit() else None)
+        if not self.cpu_nstats:
+            return []
+
+        if key not in self.cpu_nstats[0]:
+            return []
 
-        return nstats
+        return [int(cpu[key], 16) for cpu in self.cpu_nstats]
```

Comment:
Just came into my mind, it would be better to shift the calculating/casting to `parse_content`, since each column must be a hex-number-string. This will save the workload and avoid calculating/casting it for every calling of `per_cpu_nstat`.

How do you think?

~~~
     def parse_content(self, contents):
         nstats = []
          for line in contents:
              stats = [int(st,16) for st in line.split(None)]
              self.cpu_nstats.append({'packet_process': stats[0],
                              'packet_drops': stats[1],
                              'time_squeeze': stats[2],
                              'cpu_collision': stats[8],
                              'received_rps': stats[9],
                              'flow_limit_count': stats[10]})
~~~
Or, if you want skip the first column "packet_process" (I do not familiy with it)
~~~
              stats = line.split(None)
              stats = [int(st,16) for st in stats[1:]]
~~~